README.md

```diff
@@ -459,6 +459,26 @@ client.buildURL(
 
 ```
 
+The custom encoder also accepts a second optional parameter `key` which allows users to modify how query parameters are encoded. This parameter does not affect the custom encoding logic of the image path.
+
+```js
+const ImgixClient = require("@imgix/js-core");
+const client = new ImgixClient({
+  domain: 'test.imgix.com',
+  secureURLToken: 'xxxxxxxx',
+});
+
+client.buildURL(
+  "https://proxy.imgix.net/image.jpg",
+  {
+    "txt": "test!(')*"
+  },
+  {
+    encoder: (value, key) => key.substr(-2) === '64' ? Base64.encodeURI(value) : value.replace(' ', "+")
+  }
+)
```

Comment:
I think we should optionally chain here, otherwise, this might throw an error when encoding the path.

```suggestion
    encoder: (value, key) => key?.substr(-2) === '64' ? Base64.encodeURI(value) : value.replace(' ', "+")
  }
)
```

Reply:
Yep good catch